qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/completions/QuteSnippetCompletionInWhenSectionTest.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.services.completions;
+
+import static com.redhat.qute.QuteAssert.SECTION_SNIPPET_SIZE;
+import static com.redhat.qute.QuteAssert.c;
+import static com.redhat.qute.QuteAssert.r;
+import static com.redhat.qute.QuteAssert.testCompletionFor;
+
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for Qute snippet completion in #when section.
+ *
+ */
+public class QuteSnippetCompletionInWhenSectionTest {
+

```

Comment:
it might also be interesting to test the case where you have:
```
{#when value}
  {#is asdf}
  |
{/when}
```
and you open completion at the cursor, since you should be able to complete it many times (this case appears to work as expected right now).